dask/array/creation.py

```diff
@@ -209,16 +202,32 @@ def full_like(a, fill_value, order="C", dtype=None, chunks=None, name=None):
     """
 
     a = asarray(a, name=False)
+    shape, chunks = _get_like_function_shapes_chunks(a, chunks, shape)
     return full(
-        a.shape,
+        shape,
         fill_value,
-        order=order,
         dtype=(dtype or a.dtype),
-        chunks=(chunks if chunks is not None else a.chunks),
+        order=order,
+        chunks=chunks,
         name=name,
     )
 
 
+def _get_like_function_shapes_chunks(a, chunks, shape):
+    """
+    Helper function for finding shapes and chunks for *_like()
+    array creation functions.
+    """
+    if shape is None:
+        shape = a.shape
+    if chunks is None:
+        if shape is None:
+            chunks = a.chunks
+        else:
+            chunks = "auto"
```

Comment:
Unfortunately this doesn't fix the failing test, but it is an improvement to the logic I think

```suggestion
    if shape is None:
        shape = a.shape
        if chunks is None:
            chunks = a.chunks
    
    
    elif chunks is None:
        chunks = "auto""
```

Reply:
👍